src/api/entityResolution/types.ts

```diff
@@ -395,3 +395,144 @@ export interface EntityResolutionMetrics {
    */
   possibleRelationships: number;
 }
+
+export interface ExplainWhyRecordParams extends IDataSourceParams {
+  /**
+   * The ID of the record to explain. It corresponds to the ID of the record node in the graph
+   * database.
+   */
+  recordId: string;
+}
+
+export interface ExplainWhyEntitiesParams extends IDataSourceParams {
+  /**
+   * The ID of the first entity to explain. It corresponds to the `entityId` property of the entity
+   * node in the graph database.
+   */
+  entityId: number;
+  /**
+   * The ID of the other related entity.
+   */
+  otherEntityId: number;
+}
+
+export interface GetEntityByIdParams extends IDataSourceParams {
+  /**
+   * The ID of the first entity to explain. It corresponds to the `entityId` property of the entity
+   * node in the graph database.
+   */
+  entityId: number;
+}
+
+export interface WhyRecord {
+  matchLevelCode: 'RESOLVED';
+  matchKey: MatchKey;
+  entityResolutionRuleCode: string;
+  matchScores: MatchScore<ScoredValue>[];
+}
+
+export interface WhyEntities {
+  matchLevelCode: 'POSSIBLY_SAME' | 'POSSIBLY_RELATED';
+  matchKey: MatchKey;
+  entityResolutionRuleCode: string;
+  matchScores: MatchScore<ScoredPair>[];
+}
+
+interface MatchKey {
+  /**
+   * The serialized form of the match key, for instance `+NAME+ADDRESS-DOB`.
+   */
+  value: string;
+  same: EntityAttributeKey[];
+  different: EntityAttributeKey[];
+  isAmbiguous: boolean;
+}
+
+interface MatchScore<T> {
+  key: EntityAttributeKey;
+  values: T[];
+}
+
+interface ScoredValue {
+  value: string;
+  score: {
+    percentage: number;
+    bucket: 'SAME' | 'CLOSE' | 'PLAUSIBLE' | 'NO_CHANCE';
+  };
+}
+
+interface ScoredPair extends ScoredValue {
+  otherValue: string;
+}
+
+type EntityAttributeKey =
+  | 'Account number'
+  | 'Address'
+  | 'Citizenship'
+  | 'Date of birth'
+  | 'Date of death'
+  | 'Driver license'
+  | 'DUNS number'
+  | 'Email'
+  | 'Facebook'
+  | 'Gender'
+  | 'Group association'
+  | 'Instagram'
+  | 'LEI number'
+  | 'LinkedIn'
+  | 'Name'
+  | 'Nationality'
+  | 'National ID'
+  | 'NPI number'
+  | 'Passport'
+  | 'Phone'
+  | 'Place of birth'
+  | 'Record type'
+  | 'Registration country'
+  | 'Registration date'
+  | 'Signal'
+  | 'Social security number'
+  | 'Skype'
+  | 'Tango'
+  | 'Tax ID'
+  | 'Telegram'
+  | 'Twitter'
+  | 'Viber'
+  | 'Website'
+  | 'Wechat'
+  | 'WhatsApp'
+  | 'Zoom room';
+
+// Internal note: this type is also used by LKE's entity resolution client
```

Comment:
I'm not convinced these comments are useful, but I'm OK if you'd like to keep them.

`LKE's entity resolution client` seems confusing to me. I suggest calling it `entity resolution client` to makes it clearer we're not talking of the LKE REST Client (the one in this repo).

Reply:
https://github.com/Linkurious/linkurious-rest-client/pull/803/commits/a1485084cad78ca13e0a956de094b34e500a1237